music21/chord/__init__.py

```diff
@@ -47,9 +47,339 @@ class ChordException(exceptions21.Music21Exception):
+class ChordBase(note.NotRest):
+    '''
+    A base class for NotRest objects that have multiple underlying structures
+    like notes or unpitched percussion.
+    '''
+    isNote = False
+    isRest = False
+
+    _DOC_ATTR = {
+        'isNote': '''
+            Boolean read-only value describing if this
+            GeneralNote object is a Note. Is False''',
+        'isRest': r'''
+            Boolean read-only value describing if this
+            GeneralNote object is a Rest. Is False
+
+            >>> c = chord.Chord()
+            >>> c.isRest
+            False
+            ''',
+        'beams': 'A :class:`music21.beam.Beams` object.',
+    }
+
+
+    # update inherited _DOC_ATTR dictionary
+    _DOC_ATTR.update(note.NotRest._DOC_ATTR)
+
+    def __init__(self, notes=None, **keywords):
+        if notes is None:
+            notes = []
+        if isinstance(notes, str):
+            if ' ' in notes:
+                notes = notes.split()
+            else:
+                notes = [notes]
+
+        # the list of pitch objects is managed by a property; this permits
+        # only updating the _chordTablesAddress when pitches has changed
+
+        # duration looks at _notes to get first duration of a pitch
+        # if no other pitches are defined
+
+        # a list of dictionaries; each storing pitch, tie, and volume objects
+        # one for each component of the chord
+        self._overrides = {}
+        # self._cache is now defined on Music21Object.
+        # self._cache = {}
+
+        self._notes: List[note.NotRest] = []
+        # here, pitch and duration data is extracted from notes
+        # if provided
+
+        super().__init__(**keywords)
+
+        # inherit Duration object from GeneralNote
+        # keep it here in case we have no notes
+        # self.duration = None  # inefficient, since note.Note.__init__ set it
+        # del self.pitch
+        durationKeyword = None
+        if 'duration' in keywords:
+            durationKeyword = keywords['duration']
+
+        durationKeyword = self._add_core_or_init(notes, useDuration=durationKeyword)
+
+        if durationKeyword is not None:
+            self.duration = durationKeyword
+        elif 'type' in keywords or 'quarterLength' in keywords:  # dots dont cut it
+            self.duration = duration.Duration(**keywords)
+
+        # elif len(notes) > 0:
+        #     for thisNote in notes:
+        #         # get duration from first note
+        #         # but should other notes have the same duration?
+        #         self.duration = notes[0].duration
+        #         break
+
+        if 'beams' in keywords:
+            self.beams = keywords['beams']
+        else:
+            self.beams = beam.Beams()
+
+    def __eq__(self, other):
+        '''
+        True if the Chord passes all `super()`
+        equality tests and the pitches are the same
+        (possibly in a different order)
```

Comment:
This probably needs to be changed for BaseChord.  These docs look more like they belong in Chord.__eq__

Reply:
I rewrote it slightly.